src/components/routes/Account/FirebaseEmailHandler/PasswordReset.tsx

```diff
@@ -0,0 +1,149 @@
+import * as React from 'react';
+import AudioLoading from 'shared/loading/AudioLoading';
+import { auth } from 'utils/firebase';
+import Button from 'shared/Button';
+import InputWrapper from 'shared/InputWrapper';
+import BeeLink from 'shared/BeeLink';
+
+interface State {
+  isSubmitting: boolean;
+  hasError: boolean;
+  showPasswordResetForm: boolean;
+  showPasswordResetSuccess: boolean;
+  successMessage: string;
+  errorMessage: string;
+  password: string;
+}
+
+interface Props {
+  oobCode: string; //firebase code from email
+}
+
+export default class PasswordReset extends React.Component<Props> {
+  readonly state: State = {
+    isSubmitting: false,
+    hasError: false,
+    showPasswordResetForm: true,
+    showPasswordResetSuccess: false,
+    errorMessage: '',
+    successMessage: '',
+    password: '',
+  };
+
+  componentDidMount() {
+    const { oobCode } = this.props;
+
+    if (!oobCode) {
+      alert('Error, no oobCode defined.');
+      return;
+    }
+
+    // @see
+    // https://firebase.google.com/docs/auth/custom-email-handler
+    this.handleResetPassword(oobCode);
+  }
+
+  handleFormChange = (event: React.ChangeEvent<HTMLInputElement>): void => {
+    const { name, value } = event.target;
+    return;
+    this.setState({
+      [name]: value,
+    });
+  };
+
+  // @see
+  // https://firebase.google.com/docs/auth/custom-email-handler
+  handleResetPassword(actionCode: string) {
+    // Verify the password reset code
+    // is valid.
+    auth
+      .verifyPasswordResetCode(actionCode)
+      .then(() => {
+        this.setState({ showPasswordResetForm: true });
+      })
+      .catch(() => {
+        this.setState({
+          hasError: true,
+          errorMessage: 'Invalid or expired action code.Ask user to try to reset the password again.',
+        });
+      });
+  }
+
+  handlePasswordResetSubmit(event: React.FormEvent) {
+    event.preventDefault();
+
+    // Save the new password.
+    auth
+      .confirmPasswordReset(this.props.oobCode, this.state.password)
+      .then(() => {
+        this.setState({ showPasswordResetSuccess: true });
+      })
+      .catch(() => {
+        this.setState({
+          hasError: true,
+          errorMessage: 'Password is too weak or the password reset link has expired.',
```

Comment:
maybe we should tell them our criteria for passwords? something like `Please make sure your password is at least 8 characters long.` so they know if it's their password that's wrong or the link has expired